### ISSUE TITLE
Issues with starting embedded server

### ISSUE THREAD

**Issue by hrstoyanov** [orientechnologies/orientdb#8285]
### OS: Windows 7 64-bit

 
I am trying to start an Embedded server (over a database that I have previously created with ```console.bat```) with his piece of code:

```java
        //....
        log.info(()->"ODB-JCA resource adapter starting ...");
        log.info(()->"ODB-JCA server configuration: " + getEmbeddedServerConfiguration());
        log.info(()->"ODB-JCA server home: " + getOrientdbHome());
        if (embeddedServerConfiguration != null) 
        try {
            embeddedServer = OServerMain.create(true);
            embeddedServer.startup(new File(embeddedServerConfiguration));
            embeddedServer.activate();

            log.info(()->"ODB-JCA database directory: " + embeddedServer.getDatabaseDirectory());
            log.info(()->"ODB-JCA server id: " + embeddedServer.getServerId());
            log.info(()->"ODB-JCA server activated: " + embeddedServer.isActive());

            Map<String, String> storageNames = embeddedServer.getAvailableStorageNames();
            if (storageNames != null) {
                storageNames.forEach((String k, String v) -> log.info(()->"ODB-JCA storage: " + k + ", " + v));
            }

            Map<String, Class<? extends ONetworkProtocol>> networkProtocols = embeddedServer.getNetworkProtocols();
            if (networkProtocols != null) {
                networkProtocols.forEach((String k, Class<? extends ONetworkProtocol> v) -> {
                    log.info(()->"ODB-JCA network protocol name: " + k + ", " + v.getSimpleName());
                    OServerNetworkListener l = embeddedServer.getListenerByProtocol(v);
                    log.info(()->"ODB-JCA protocol listener inbound addr: " + l.getInboundAddr() + ", active:" + l.isActive() + ", alive:" + l.isAlive());
                });
            }

            log.info(()->"ODB-JCA resource adapter successfully started");

        } catch (Exception e) {
            log.log(Level.SEVERE,"ODB-JCA resource adapter failed to start", e);
            throw new ResourceAdapterInternalException(e);
        }
```
## Expected behavior 
I expected the server to start with no issues. 

## Actual behavior 
Instead, this call ```embeddedServer.activate();``` bombs with the below exception - __why is the server trying to create db system files when the database already exists__? Also, looking at the database folder, I do see a lot of files, but not __OSystem__ ?
```
Caused by: com.orientechnologies.orient.core.exception.OStorageExistsException: Cannot create new storage 'plocal:C:\temp\my.db/OSystem' because it already exists
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.create(OAbstractPaginatedStorage.java:456)
        at com.orientechnologies.orient.core.storage.impl.local.paginated.OLocalPaginatedStorage.create(OLocalPaginatedStorage.java:126)
        at com.orientechnologies.orient.core.db.OrientDBEmbedded.internalCreate(OrientDBEmbedded.java:251)
        at com.orientechnologies.orient.core.db.OrientDBEmbedded.create(OrientDBEmbedded.java:209)

```
When I comment out ```//embeddedServer.activate();``` (although the ODB documentation clearly shows it is needed for embedded server) I get another exception:
```
Caused by: java.lang.NullPointerException
        at com.orientechnologies.orient.server.OServer.getServerId(OServer.java:249)
```
Do you guys inspect the codebase with something like [FindBugs](http://findbugs.sourceforge.net/) - such NPEs are pretty easy to catch?

## Steps to reproduce
1. Create an empty database
2. Configure an embedded server for work with it
```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<orient-server>
    <handlers>
        <handler class="com.orientechnologies.orient.server.handler.OServerSideScriptInterpreter">
            <parameters>
                <parameter value="true" name="enabled"/>
                <parameter value="SQL" name="allowedLanguages"/>
            </parameters>
        </handler>
    </handlers>
    <network>
        <protocols>
            <protocol
                    implementation="com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary"
                    name="binary"/>
        </protocols>
        <listeners>
            <listener protocol="binary" socket="default" port-range="3424-3430" ip-address="0.0.0.0"/>
        </listeners>
    </network>

    <users>
        <user resources="*" password="admin" name="admin"/>
    </users>

    <properties>
        <entry value="../../../../../../my.db" name="server.database.path"/>
        <entry value="1" name="db.pool.min"/>
        <entry value="50" name="db.pool.max"/>
        <entry value="true" name="profiler.enabled"/>
        <entry value="info" name="log.console.level"/>
        <entry value="info" name="log.file.level"/>
    </properties>

    <isAfterFirstTime>true</isAfterFirstTime>
```
3. Start an embedded instance as described above

**Comment by hrstoyanov** [comment]
![image](https://user-images.githubusercontent.com/5327044/40399208-d9d3d332-5df0-11e8-9bd2-2415d69ee29a.png)

Looking at the ``` OAbstractPaginatedStorage.create(...)``` there seems to be some confusion as to what ```exists()``` checks - it only verifies the db root folder. Could this be the issue?

**Comment by hrstoyanov** [comment]
I just tested with OrientDB 3.0.1 - issue still there.

**Comment by tglman** [comment]
Hi @hrstoyanov,

Are you giving a custom path for the "server.database.path" or are you using a custom storage path ?  probably would be easier if you can share the orientdb-server-config.xml that you are using.

Another issue could be in the path management, could you try to replace `\\` with `/` in the path, just to exclude a case.

Regards

**Comment by hrstoyanov** [comment]
Hi @tglman. Indeed, I am using ```server.database.path``` i my server configuration file:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<orient-server>
    <handlers>
        <handler class="com.orientechnologies.orient.server.handler.OServerSideScriptInterpreter">
            <parameters>
                <parameter value="true" name="enabled"/>
                <parameter value="SQL" name="allowedLanguages"/>
            </parameters>
        </handler>
    </handlers>
    <network>
        <protocols>
            <protocol
                    implementation="com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary"
                    name="binary"/>
        </protocols>
        <listeners>
            <listener protocol="binary" socket="default" port-range="3424-3430" ip-address="0.0.0.0"/>
        </listeners>
    </network>

    <users>
        <user resources="*" password="admin" name="admin"/>
    </users>

    <properties>
        <entry value="../../../../../../my.db" name="server.database.path"/>
        <entry value="1" name="db.pool.min"/>
        <entry value="50" name="db.pool.max"/>
        <entry value="true" name="profiler.enabled"/>
        <entry value="info" name="log.console.level"/>
        <entry value="info" name="log.file.level"/>
    </properties>

    <isAfterFirstTime>true</isAfterFirstTime>
```

**Comment by markodjurovic** [comment]
@hrstoyanov Thank you for reporting this issue and on detailed explanation. Fix will be part of 3.0.2 version, also it is already merged into development branch
